### PR TITLE
Update on the PrestaShop DevDocs about the hook on the moderns pages

### PR DESCRIPTION
**Pull request by Antoine-OH**
Hello I want to contribute to the community so here is my proposition. I was learning the prestashop 1.7.6 with this doc but it's not up to date so the part on the hooks on moderne pages wasn't working. I had to change the namespace and to create the method getRepository to create the repository and load the service without this, the erreur : You have requested a non-existent service is thrown

### CODE REVIEW

**Comment by Antoine-OH** [comment]
Warning, i didn't changed the <Using the Symfony components to create an XML export file> so you will have to change it yourself, i only corrected the first part about the dumb of the products

**Comment by Antoine-OH** [comment]
Okay my bad i failed somewhere in my code, in the repository I left $databasePrefix instead of $dbPrefix like its declared in my service, you have to replace all databasePrefixby dbPrefixe (line 106/108/110 and 120) sorry i'm new and never contributed to projects so I don't know if can modify my pull request etc...

**Comment by matks** [comment]
Hi @Antoine-OH thanks for your motivation to contribute 😉 

You can modify your PR of course 😉 either by modifying the branch on your fork or by viewing the files you changed and using the "edit" button in github interface

**Comment by Antoine-OH** [comment]
Ty i made the changes, by the way the end of the doc about xml files don't work for me and i don't know how to solve it.

File: src/content/1.7/modules/concepts/hooks/use-hooks-on-modern-pages.md [hunk 1]
```diff
@@ -45,41 +45,7 @@ Accessing the Product Catalog page in *debug mode* we can access the list of ava
 
 As we need to act on Dashboard but after the header, in the icons toolbar (with others export options) `hookdisplayDashboardToolbarIcons` sounds like the hook we are looking for.
 
-### Second step: create and register the Hook
```
Comment (jolelievre) [inline src/content/1.7/modules/concepts/hooks/use-hooks-on-modern-pages.md:48]: The step by step layout is easier for beginner developers

File: src/content/1.7/modules/concepts/hooks/use-hooks-on-modern-pages.md [hunk 2]
```diff
@@ -117,49 +86,138 @@ class ProductRepository
...
+
+use PrestaShop\Module\Foo\Repository\ProductRepository;
+use PrestaShop\PrestaShop\Adapter\SymfonyContainer;
+
+class Foo extends Module
 {
-    if ($this->isSymfonyContext() && $hookParams['route'] === 'admin_product_catalog') {
-        $products = $this->get('product_repository')->findAllByLangId(1);
-        dump($products);
+    public function __construct()
```
Comment (jolelievre) [inline src/content/1.7/modules/concepts/hooks/use-hooks-on-modern-pages.md:139]: This code is now a bit long, you could reduce it using `...` inside the class
The idea is not to provide a fully functional class to copy, but to emphasize the code related to what you are explaining

So in this part the constructor, and the install/uninstall methods are less relevant. You should focus on the hook code, by the way nice display of the fallback service creation

**Comment by Progi1984** [comment]
Ping @Antoine-OH

**Comment by matks** [comment]
ping @Antoine-OH are you still with us ? 😅

**Comment by matks** [comment]
Unfortunatey author does not answer since 21 days :/ so I close the PR

If however you come back dont worry we can reopen the PR 😄

**Pull request closed without merging**